java/ql/src/semmle/code/java/dataflow/internal/TaintTrackingUtil.qll

```diff
@@ -722,6 +749,56 @@ class ObjectOutputStreamVar extends LocalVariableDecl {
   }
 }
 
+/** Flow through string formatting. */
+private predicate formatStep(Expr tracked, Expr sink) {
```

Comment:
@aschackmull These predicates and classes implement the same pattern as https://github.com/github/codeql/blob/be07d27a4c3594d5de4f77c17de8e6a7d874f4ef/java/ql/src/semmle/code/java/dataflow/internal/TaintTrackingUtil.qll#L712-L750 . 

Do you know of a better way of implementing this? Ideally we would capture the idea in a generalised predicate that could be re-used.

Reply:
This kind of step is a bit more ad-hoc than most of the others we include, so without some examples of real code making use of such a pattern that would otherwise lead to FNs, then I'd lean towards excluding this step.  I'd expect the `serializationStep` to be mostly irrelevant, but I can imagine that it has probably been added a long time ago to improve coverage in some artificial test suite.

Reply:
This query: https://lgtm.com/query/1827343111074051413/ shows that it's not an uncommon pattern for a `Formatter` to be constructed with a `StringBuilder` which later has `toString` called on it, so it's not hard to imagine that it could otherwise lead to FNs.